Tasks/AzureFileCopy/AzureFileCopyJob.ps1

```diff
@@ -62,14 +62,18 @@ param (
 
     Write-Verbose "Initiating copy on $fqdn "
 
+    $storageAccountInfo = Get-AzureStorageAccount -StorageAccountName $storageAccount
+	$stoargeAccountEnpoints = $storageAccountInfo.Endpoints
+	$blobStorageURI = $stoargeAccountEnpoints[0]+$containerName+"/"+$blobPrefix
+	
     [String]$copyToAzureMachinesBlockString = [string]::Empty
     if([string]::IsNullOrWhiteSpace($additionalArguments))
     {
-        $copyToAzureMachinesBlockString = "Copy-ToAzureMachines -MachineDnsName `$fqdn -StorageAccountName `$storageAccount -ContainerName `$containerName -SasToken `$sasToken -DestinationPath `$targetPath -Credential `$credential -AzCopyLocation `$azCopyLocation -WinRMPort $winRMPort $cleanTargetPathOption $skipCACheckOption $httpProtocolOption $enableDetailedLoggingOption"
+        $copyToAzureMachinesBlockString = "Copy-ToAzureMachines -MachineDnsName `$fqdn -StorageAccountName `$storageAccount -ContainerName `$containerName -SasToken `$sasToken -DestinationPath `$targetPath -Credential `$credential -AzCopyLocation `$azCopyLocation -BlobStorageURI `$blobStorageURI -WinRMPort $winRMPort $cleanTargetPathOption $skipCACheckOption $httpProtocolOption $enableDetailedLoggingOption"
     }
     else
     {
-        $copyToAzureMachinesBlockString = "Copy-ToAzureMachines -MachineDnsName `$fqdn -StorageAccountName `$storageAccount -ContainerName `$containerName -SasToken `$sasToken -DestinationPath `$targetPath -Credential `$credential -AzCopyLocation `$azCopyLocation -AdditionalArguments `$additionalArguments -WinRMPort $winRMPort $cleanTargetPathOption $skipCACheckOption $httpProtocolOption $enableDetailedLoggingOption"
+        $copyToAzureMachinesBlockString = "Copy-ToAzureMachines -MachineDnsName `$fqdn -StorageAccountName `$storageAccount -ContainerName `$containerName -SasToken `$sasToken -DestinationPath `$targetPath -Credential `$credential -AzCopyLocation `$azCopyLocation -AdditionalArguments `$additionalArguments -BlobStorageURI `$blobStorageURI -WinRMPort $winRMPort $cleanTargetPathOption $skipCACheckOption $httpProtocolOption $enableDetailedLoggingOption"
```

Comment:
Couple things for you to consider:
- Is there a reason for the indirection here? I.e. building up a string, converting to a scriptblock, then calling invoke command on the scriptblock?
- Does this suffer from problems with special characters not being escaped properly?
- Is a goal here to dynamically expand AdditionalArguments? If so, it makes me wonder if just that one piece should be dynamically expanded. $ExecutionContext.InvokeCommand.ExpandString might be more appropriate for just that one parameter - I don't know.
- I'm also wondering if all the string->scriptblock effort is partly due to the CleanTargetPath and EnableDetailedLogging switches. If so, it looks like those could be more appropriately passed using the syntax `-CleanTargetPath:$true` or `-CleanTargetPath:$false` which enables you to avoid the string->scriptblock business.
- Why capture the result of Invoke-Command? It gets in the way of the streaming output - i.e. causes all output to be batched until the entire command finishes.

I don't see any comments explaining anything in the file, so I'll assume my suggestions haven't been considered before ;)